pandas/core/ops/array_ops.py

```diff
@@ -132,3 +152,171 @@ def na_arithmetic_op(left, right, op, str_rep, eval_kwargs):
         result = masked_arith_op(left, right, op)
 
     return missing.dispatch_fill_zeros(op, left, right, result)
+
+
+def arithmetic_op(left, right, op, str_rep, eval_kwargs):
+
+    from pandas.core.ops import (
+        maybe_upcast_for_op,
+        should_extension_dispatch,
+        dispatch_to_extension_op,
+    )
+
+    keep_null_freq = isinstance(
+        right,
+        (
+            ABCDatetimeIndex,
+            ABCDatetimeArray,
+            ABCTimedeltaIndex,
+            ABCTimedeltaArray,
+            Timestamp,
+        ),
+    )
+
+    lvalues = extract_array(left, extract_numpy=True)
```

Comment:
So we're completely done with deferring via NotImplemented by the time we get here?

Reply:
Yes.  So for e.g. PandasArray we will be able to use these directly only after #28037 is in.